src/client/driver/command-executors/client-functions/selector-executor/filter.js

```diff
@@ -54,15 +54,16 @@ class SelectorFilter {
             else
                 filtered = filtered.length;
         }
-        else if (options.collectionMode) {
-            if (options.index !== null) {
-                const nodeOnIndex = this.getNodeByIndex(filtered, options.index);
+        else {
+            if (options.collectionMode) {
```

Comment:
we can merge two this conditions
`if(options.collectionMode && options.index !== null)`

Reply:
we can't. If `options.collectionMode === true` and `options.index === null` we should not modify `filtered` collection

Reply:
You're right)